app.js

```diff
@@ -1,18 +1,94 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
-
-// require spotify-web-api-node package here:
+const express = require("express");
+const hbs = require("hbs");
+const SpotifyWebApi = require("spotify-web-api-node");
 
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
+
+app.use(express.urlencoded({ extended: true }));
 
 // setting the spotify-api goes here:
 
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
+
 // Our routes go here:
 
-app.listen(3000, () => console.log('My Spotify project running on port 3000 🎧 🥁 🎸 🔊'));
+app.get("/", (req, res) => {
+  res.render("search");
+});
+
+app.get("/artist-search", async (req, res) => {
+  const Sresponse = await spotifyApi
+    .searchArtists(req.query.name)
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+  const atistsItems = Sresponse.body.artists.items;
+  // console.log("The received data from the API: ", Sresponse.body);
+
+  // for (let item of atistsItems) {
+  //   console.log(item);
+  // }
+
+  const data = { doctitle: "Artists", artists: atistsItems };
+
+  res.render("artist-search-results", data);
+});
+
+app.get("/albums/:artistId", async (req, res) => {
+
+  const artistResponse = await spotifyApi.getArtist(req.params.artistId)
+  .catch((err) =>
+    console.log("The error while searching artists occurred: ", err)
+  );
+
+  const Sresponse = await spotifyApi
+    .getArtistAlbums(req.params.artistId)
+    .catch((err) =>
```

Comment:
you start duplicating this .catch handler -> maybe instead use try catch and have one error handling for this whole block? 

also in doing so you would fix that certain things that are still not properly error handled -> i.e. when Sresponse.body is undefined for example, because the API doesn't return the correct thing for some reason your could would kill the server.